2.16.3-20221209 incompatible with 6.4.x kernels

Hi,

```
# dkms install -m r8152 -v 2.16.3-20221209 -k `uname -r`
Sign command: /lib/modules/6.4.10-100.fc37.x86_64/build/scripts/sign-file
Signing key: /var/lib/dkms/mok.key
Public certificate (MOK): /var/lib/dkms/mok.pub

Building module:
Cleaning build area...
make -j12 KERNELRELEASE=6.4.10-100.fc37.x86_64 -C /lib/modules/6.4.10-100.fc37.x86_64/build M=/var/lib/dkms/r8152/2.16.3-20221209/build....(bad exit status: 2)
Error! Bad return status for module build on kernel: 6.4.10-100.fc37.x86_64 (x86_64)
Consult /var/lib/dkms/r8152/2.16.3-20221209/build/make.log for more information.
```

     and in dkms make.log

```
# cat /var/lib/dkms/r8152/2.16.3-20221209/build/make.log
DKMS make.log for r8152-2.16.3-20221209 for kernel 6.4.10-100.fc37.x86_64 (x86_64)
Sun Aug 13 06:03:33 PM CEST 2023
make: Entering directory '/usr/src/kernels/6.4.10-100.fc37.x86_64'
  CC [M]  /var/lib/dkms/r8152/2.16.3-20221209/build/r8152.o
/var/lib/dkms/r8152/2.16.3-20221209/build/r8152.c: In function ‘r8152_csum_workaround’:
/var/lib/dkms/r8152/2.16.3-20221209/build/r8152.c:2074:24: error: implicit declaration of function ‘skb_gso_segment’; did you mean ‘skb_gso_reset’? [-Werror=implicit-function-declaration]
 2074 |                 segs = skb_gso_segment(skb, features);
      |                        ^~~~~~~~~~~~~~~
      |                        skb_gso_reset
/var/lib/dkms/r8152/2.16.3-20221209/build/r8152.c:2074:22: warning: assignment to ‘struct sk_buff *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2074 |                 segs = skb_gso_segment(skb, features);
      |                      ^
cc1: some warnings being treated as errors
make[1]: *** [scripts/Makefile.build:252: /var/lib/dkms/r8152/2.16.3-20221209/build/r8152.o] Error 1
make: *** [Makefile:2050: /var/lib/dkms/r8152/2.16.3-20221209/build] Error 2
make: Leaving directory '/usr/src/kernels/6.4.10-100.fc37.x86_64'

```

Any chance of a fix?   The default driver version in FC37 is rather unstable.

--Tore

--- a/r8152.c
+++ b/r8152.c
@@ -25,6 +25,12 @@
 #include <linux/ip.h>
 #include <linux/ipv6.h>
 #include <net/ip6_checksum.h>
+// Linux 6.4.10 added net/gso.h
+#if defined __has_include
+#if __has_include (<net/gso.h>)
+#include <net/gso.h>
+#endif
+#endif
 #include <linux/usb/cdc.h>
 #include <linux/suspend.h>
 #include <linux/atomic.h>

